tests/test_minimizer.py

```diff
@@ -62,3 +72,488 @@ def test_minimizer_fit(data):
+
+    preprocessor = ColumnTransformer(
+        transformers=[
+            ("num", numeric_transformer, numeric_features),
+            ("cat", categorical_transformer, categorical_features),
+        ]
+    )
+    encoded = preprocessor.fit_transform(X)
+    base_est = DecisionTreeClassifier()
+    base_est.fit(encoded, y)
+    predictions = base_est.predict(encoded)
+    # Append classifier to preprocessing pipeline.
+    # Now we have a full prediction pipeline.
+    gen = GeneralizeToRepresentative(base_est, features=features, target_accuracy=0.5,
+                                     categorical_features=categorical_features, quasi_identifiers=QI)
+    gen.fit(X, predictions)
+    transformed = gen.transform(X)
+    print(X)
+    print(transformed)
+
+
+def test_minimize_ndarray_iris():
+    features = ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']
+    (x_train, y_train), _ = get_iris_dataset()
+
+    QI = [0, 2]
+    model = DecisionTreeClassifier()
+    model.fit(x_train, y_train)
+    pred = model.predict(x_train)
+
+    gen = GeneralizeToRepresentative(model, target_accuracy=0.3, features=features, quasi_identifiers=QI)
+    gen.fit(x_train, pred)
+    transformed = gen.transform(x_train)
+    print(transformed)
+
+
+def test_minimize_pandas_adult():
+    (x_train, y_train), _ = get_adult_dataset()
+    x_train = x_train.head(12000)
+    y_train = y_train.head(12000)
+
+    features = ['age', 'workclass', 'education-num', 'marital-status', 'occupation', 'relationship', 'race', 'sex',
+                'capital-gain', 'capital-loss', 'hours-per-week', 'native-country']
+
+    categorical_features = ['workclass', 'marital-status', 'occupation', 'relationship', 'race', 'sex',
+                            'hours-per-week', 'native-country']
+
+    QI = ['age', 'workclass', 'education-num', 'marital-status', 'occupation', 'relationship', 'race', 'sex',
+          'native-country']
+
+    numeric_features = [f for f in features if f not in categorical_features]
+    numeric_transformer = Pipeline(
+        steps=[('imputer', SimpleImputer(strategy='constant', fill_value=0))]
+    )
+    categorical_transformer = OneHotEncoder(handle_unknown="ignore", sparse=False)
+    preprocessor = ColumnTransformer(
+        transformers=[
+            ("num", numeric_transformer, numeric_features),
+            ("cat", categorical_transformer, categorical_features),
+        ]
+    )
+    encoded = preprocessor.fit_transform(x_train)
+    base_est = DecisionTreeClassifier()
+    base_est.fit(encoded, y_train)
+    predictions = base_est.predict(encoded)
+
+    gen = GeneralizeToRepresentative(base_est, target_accuracy=0.7, features=features,
+                                     categorical_features=categorical_features, quasi_identifiers=QI)
+    gen.fit(x_train, predictions)
+    transformed = gen.transform(x_train)
+    print(gen.generalizations_)
+
+
+def test_experiment_adult():
```

Comment:
I think these could be removed, leave them only if they add something as far as tested functionality. If it's just a repeat of existing tests they are not needed.
If you leave them, they need to perform asserts instead of prints.